### PR TITLE
transform: Add case-changing transforms

### PR DESCRIPTION
**Pull request by jlucovsky**
DRAFT: suricata-verify tests will be added soon

Add the transforms for changing the case of a buffer. 

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: (6439)[https://redmine.openinfosecfoundation.org/issues/6439]

Describe changes:
- Add to_lowercase and to_uppercase transforms

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1460
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: src/detect-transform-casechange.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+static int DetectTransformToLowerSetup(DetectEngineCtx *, Signature *, const char *);
+static void DetectTransformToLower(InspectionBuffer *buffer, void *options);
+void DetectTransformToLowerRegister(void);
+
+/* to_uppercase */
+static int DetectTransformToUpperSetup(DetectEngineCtx *, Signature *, const char *);
+static void DetectTransformToUpper(InspectionBuffer *buffer, void *options);
+void DetectTransformToUpperRegister(void);
+
+void DetectTransformToUpperRegister(void)
```
Comment (catenacyber) [inline src/detect-transform-casechange.c:40]: never called

**Comment by catenacyber** [comment]
Missing SV test ;-p

**Comment by victorjulien** [comment]
I think we should reject a sig like `http.uri; to_uppercase; content:"lowercase";` or `http.uri; to_lowercase; content:"UPPERCASE";`

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16438

**Comment by jlucovsky** [comment]
> I think we should reject a sig like `http.uri; to_uppercase; content:"lowercase";` or `http.uri; to_lowercase; content:"UPPERCASE";`

Good point. Would you like that addressed in this PR or a separate one?

**Comment by jlucovsky** [comment]
> I think we should reject a sig like `http.uri; to_uppercase; content:"lowercase";` or `http.uri; to_lowercase; content:"UPPERCASE";`

Good point. I'll update the PR with additional validation checks.

**Comment by jlucovsky** [comment]
Continued in #9786

**Pull request closed without merging**